### ISSUE TITLE
Cannot get ggeffects for glm.nb and glm with contiunous covariate

### ISSUE THREAD

**Issue by asierrl** [strengejacke/ggeffects#44]
I'm using ggeffects to get marginal effects plotted from different glmer and glm models and I've found out ggeffects gives  the following result when asking for a marginal effect of one of the continuous variables with glm.nb or glm models:

[1] x         predicted conf.low  conf.high
<0 rows> (or 0-length row.names)

For example, I cannot get ggeffects for anything with anosc in this model: 
fit<- glm.nb(formula = ncone.san ~ Tratamiento * C + anosc + I(anosc^2) + 
    anosc:Tratamiento + I(anosc^2):Tratamiento + ivsc, data = datos, 
    init.theta = 1.133641349, link = log)

ggeffect(fit, c("anosc")) gives 0 rows, while ggeffect(fit, c("ivsc")) gives correct ggeffects:
\# Predicted incidents for N.sanas.sum 
\# x = ivsc 

  x predicted conf.low conf.high
 -1   442.571  254.379   769.991
  0   274.681  170.779   441.798
  1   170.480   82.409   352.674
  2   105.808   35.151   318.496
  3    65.670   14.490   297.622
  4    40.758    5.898   281.638

Both are scales variables and are numeric (class(fit$model$anosc) = class(fit$model$ivsc)=numeric). The only difference is that anosc has only four unique values, but I don't think this should make a difference.

Any hint of what's going on?
Best regards, 
Asier

**Comment by strengejacke** [comment]
Please provide a reproducible example. I can't reproduce your issue:

``` r
library("ggeffects")
library("sjmisc")
library("MASS")

data(efc)
efc$e42dep <- to_label(efc$e42dep)
fit <-
  glm.nb(
    tot_sc_e ~ neg_c_7 + I(neg_c_7 ^ 2) + neg_c_7:e42dep + I(neg_c_7 ^ 2):e42dep + c12hour + c172code,
    data = efc,
    init.theta = 1.133641349,
    link = log
  )

ggpredict(fit, "neg_c_7")
#> 
#> # Predicted counts of Services for elderly 
#> # x = Negative impact with 7 items 
#> 
#>   x predicted std.error conf.low conf.high
#>   7     0.576     0.202    0.388     0.856
#>   9     0.714     0.192    0.490     1.040
#>  11     0.879     0.200    0.595     1.300
#>  13     1.077     0.255    0.653     1.777
#>  15     1.313     0.378    0.626     2.753
#>  17     1.590     0.563    0.527     4.796
#>  19     1.916     0.804    0.396     9.271
#>  21     2.296     1.098    0.267    19.755
#>  23     2.736     1.443    0.162    46.259
#>  28     4.140     2.523    0.029   581.423
#> 
#> Adjusted for:
#> *   e42dep = independent
#> *  c12hour =       42.42
#> * c172code =        1.97
#> Standard errors are on link-scale (untransformed).

ggeffect(fit, "neg_c_7")
#> 
#> # Predicted counts of Services for elderly 
#> # x = Negative impact with 7 items 
#> 
#>   x predicted std.error conf.low conf.high
#>   6     0.517     0.211    0.341     0.782
#>   8     0.642     0.196    0.437     0.942
#>  10     0.793     0.192    0.544     1.155
#>  12     0.974     0.220    0.633     1.499
#>  14     1.190     0.308    0.650     2.180
#>  16     1.446     0.463    0.582     3.589
#>  20     2.099     0.945    0.329    13.409
#>  22     2.508     1.264    0.210    29.989
#>  24     2.981     1.634    0.121    73.644
#>  28     4.140     2.523    0.029   585.672
#> Standard errors are on link-scale (untransformed).

ggpredict(fit, c("neg_c_7", "e42dep"))
#> 
#> # Predicted counts of Services for elderly 
#> # x = Negative impact with 7 items 
#> 
#> # independent
#>   x predicted std.error conf.low conf.high
#>   7     0.576     0.202    0.388     0.856
#>  10     0.793     0.192    0.544     1.155
#>  14     1.190     0.308    0.650     2.178
#>  17     1.590     0.563    0.527     4.796
#>  20     2.099     0.945    0.330    13.373
#>  28     4.140     2.523    0.029   581.423
#> 
#> # slightly dependent
#>   x predicted std.error conf.low conf.high
#>   7     0.721     0.122    0.568     0.916
#>  10     0.853     0.100    0.701     1.037
#>  14     0.829     0.141    0.629     1.093
#>  17     0.673     0.236    0.424     1.068
#>  20     0.465     0.403    0.211     1.024
#>  28     0.079     1.198    0.008     0.822
#> 
#> # moderately dependent
#>   x predicted std.error conf.low conf.high
#>   7     0.716     0.125    0.560     0.915
#>  10     0.900     0.086    0.760     1.066
#>  14     1.012     0.091    0.847     1.209
#>  17     0.957     0.136    0.734     1.249
#>  20     0.802     0.247    0.494     1.301
#>  28     0.275     0.849    0.052     1.455
#> 
#> # severely dependent
#>   x predicted std.error conf.low conf.high
#>   7     0.730     0.154    0.539     0.988
#>  10     0.986     0.095    0.819     1.188
#>  14     1.293     0.087    1.090     1.533
#>  17     1.436     0.098    1.185     1.739
#>  20     1.465     0.124    1.150     1.868
#>  28     1.026     0.395    0.473     2.227
#> 
#> Adjusted for:
#> *  c12hour = 42.42
#> * c172code =  1.97
#> Standard errors are on link-scale (untransformed).

ggeffect(fit, c("neg_c_7", "e42dep"))
#> 
#> # Predicted counts of Services for elderly 
#> # x = Negative impact with 7 items 
#> 
#> # independent
#>   x predicted std.error conf.low conf.high
#>   6     0.517     0.211    0.341     0.782
#>  10     0.793     0.192    0.544     1.155
#>  14     1.190     0.308    0.650     2.180
#>  16     1.446     0.463    0.582     3.589
#>  20     2.099     0.945    0.329    13.409
#>  28     4.140     2.523    0.029   585.672
#> 
#> # slightly dependent
#>   x predicted std.error conf.low conf.high
#>   6     0.658     0.146    0.494     0.877
#>  10     0.853     0.100    0.701     1.038
#>  14     0.829     0.141    0.629     1.093
#>  16     0.734     0.196    0.500     1.080
#>  20     0.465     0.403    0.210     1.026
#>  28     0.079     1.198    0.007     0.825
#> 
#> # moderately dependent
#>   x predicted std.error conf.low conf.high
#>   6     0.645     0.154    0.477     0.872
#>  10     0.900     0.086    0.760     1.067
#>  14     1.012     0.091    0.846     1.209
#>  16     0.988     0.114    0.790     1.236
#>  20     0.802     0.247    0.494     1.302
#>  28     0.275     0.849    0.052     1.458
#> 
#> # severely dependent
#>   x predicted std.error conf.low conf.high
#>   6     0.648     0.186    0.450     0.934
#>  10     0.986     0.095    0.819     1.188
#>  14     1.293     0.087    1.090     1.534
#>  16     1.399     0.093    1.165     1.681
#>  20     1.465     0.124    1.149     1.868
#>  28     1.026     0.395    0.472     2.229
#> Standard errors are on link-scale (untransformed).
```

**Comment by strengejacke** [comment]
_bump_

**Comment by asierrl** [comment]
Odd enough, when preparing the reproducible example I found that the issue only appears with the full resolution data (as shown below). I initially copied the data from the default output in R Studio, 7 digits, and the plot came out without problem. When I use the maximum number of 22 digits to enter my data, the plot for one of the covariates are not shown. Here you have the reproducible example. It's my first one, so, please help me improve it if needed.
``` r
library(MASS)
library(ggeffects)
y <- c(180,111, 0, 60, 317, 224, 76, 760, 1, 64, 80, 50, 147,  324, 149, 487, 
       222, 222, 31, 2,  131,  198,  256,  364,  544,  789,  4,  113,  52, 375, 
       444,  239, 1033,  303,  129,  118,  210,  99, 398,  101, 291, 154, 244, 519, 
       99, 0,  13, 5,  416,  489,  462,  244,  74, 276, 1270,  81, 375, 1254, 36,  368, 
       114, 89,  248,  115)
trat<- c("FERTIL", "FERTIL", "FERTIL", "CTRL", "CTRL", "CTRL", "CTRL", "FERTIL", "FERTIL", "FERTIL", "FERTIL",
         "FERTIL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "CTRL", "CTRL", "FERTIL", "FERTIL", "FERTIL", "FERTIL",
         "FERTIL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "CTRL", "CTRL", "CTRL", "FERTIL", "FERTIL", "FERTIL", 
         "FERTIL", "FERTIL", "CTRL", "CTRL", "FERTIL", "FERTIL", "CTRL", "CTRL", "CTRL", "CTRL", "CTRL", 
         "CTRL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "CTRL", "CTRL", "CTRL", "CTRL", "CTRL", 
         "CTRL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "FERTIL", "CTRL", "CTRL", "CTRL")
C<-c("X2062", "X2062", "X1002", "X1002", "X1002", "X1002", "X2062", "X1002", "X1002", "X1002", "X1043", "X1043", 
     "X1002", "X1002", "X1043", "X1043", "X1043", "X1043", "X1043", "X1043", "X1043", "X2062", "X2062", "X2062", 
     "X2062", "X2062", "X1043", "X1043", "X1043", "X1002", "X2062", "X2062", "X1002", "X1002", "X1002", "X1043", 
     "X1002", "X1002", "X2062", "X1002", "X1002", "X1043", "X1043", "X2062", "X2062", "X1043", "X1043", "X1043", 
     "X1002", "X1002", "X2062", "X2062", "X2062", "X2062", "X2062", "X1002", "X2062", "X2062", "X1043", "X1043", 
     "X1043", "X1002", "X1002", "X1043")
x1 <- c(1.0510557018036901233415, -0.2181436362234073500499, -1.4873429742505046569079,
        -1.4873429742505046569079, 1.0510557018036901233415, -0.2181436362234073500499,
        1.0510557018036901233415, -0.2181436362234073500499, -1.4873429742505046569079,
        1.0510557018036901233415, -1.4873429742505046569079, -0.2181436362234073500499,
        -1.4873429742505046569079, 1.0510557018036901233415, -1.4873429742505046569079,
        -0.2181436362234073500499, -1.4873429742505046569079, 1.0510557018036901233415,
        1.0510557018036901233415, -1.4873429742505046569079, -0.2181436362234073500499,
        -1.4873429742505046569079, -0.2181436362234073500499, 1.0510557018036901233415,
        1.0510557018036901233415, -0.2181436362234073500499, -1.4873429742505046569079,
        -0.2181436362234073500499, 1.0510557018036901233415, -1.4873429742505046569079,
        1.0510557018036901233415, -1.4873429742505046569079, 1.0510557018036901233415,
        -1.4873429742505046569079, -0.2181436362234073500499, -0.2181436362234073500499,
        -0.2181436362234073500499, 1.0510557018036901233415, 1.0510557018036901233415,
        -0.2181436362234073500499, 1.0510557018036901233415, 1.0510557018036901233415,
        -0.2181436362234073500499, 1.0510557018036901233415, -0.2181436362234073500499,
        1.0510557018036901233415, -1.4873429742505046569079, -0.2181436362234073500499,
        -0.2181436362234073500499, 1.0510557018036901233415, 1.0510557018036901233415,
        -0.2181436362234073500499, 1.0510557018036901233415, -0.2181436362234073500499,
        1.0510557018036901233415, 1.0510557018036901233415, -0.2181436362234073500499,
        1.0510557018036901233415, -1.4873429742505046569079, -0.2181436362234073500499,
        1.0510557018036901233415, 1.0510557018036901233415, -0.2181436362234073500499,
        -0.2181436362234073500499)

x2 <- c(-0.16315747994073070969989, -1.05360463900379586199563, -0.93415267997080730921766
        , -0.99252850004998782740273, 0.42735235724712267169068, -0.48798991709717626230614
        , 2.23014349851008431002697, -0.21660376882282647570399, -0.72058928240081254745775
        , 1.88298518683389759509339, -0.39593086292497708944538, 0.47803447414888039501690
        , -0.87730234961981912178430, 1.45009767915056353437819, -0.64258282192604954108361
        , -0.19048163415359734007559, -0.99134159919342790256991, 0.54366244394847862420050
        , 1.66531762304370256622121, -0.79390525090304220334758, -0.01135437096435308254594
        , -1.12494052429199431486495, -0.85482479546253875568596, -0.01159930187192333150836
        , 0.75465437362335030524463, -0.54144904373718538703741, -0.85598224435921810293593
        , -0.44893610617967599774047, 1.63402389894952282389795, -1.09431441740485424496399
        , 1.22772575275634254765578, -0.88120822563015821504706, 2.18965726407797056651816
        , -0.88830486762534022204818, 0.49904785683978480914647, -0.96391262983003389575032
        , -0.84584241895151246470874, 0.01820625581143533663897, 2.04426796652258913056244
        , -0.51725662678387962500892, 0.68076969847060686547024, 0.31955573105719387028145
        , -0.76372130858969833333560, 0.48931143096917706358795, -0.63996954524487148496803
        , -0.74922815557619715232107, -1.22071057920300574117789, -1.06313667425498303842346
        , -0.39394827284068967276909, 1.14055737651990241054989, 0.08203018278837372057044
        , -0.81693043678611854119254, 0.30102881937280212554242, -0.71118362891575670481359
        , 0.16401615885635717484448, 0.07371942371769488189237, -0.34288372588738608159886
        , 2.66190601319726560802792, -0.71051489168642878446747, 0.02034948571161497923865
        , 1.92939368169321356916157, 0.75986042229322825480864, -0.70121494097762349095859
        , -0.05413653704868248706106)

reprexdata<- data.frame(y, trat, C, x1, x2)
reprexmodel <- glm.nb(y~trat*C + x1+ I(x1^2) + x1:trat + I(x1^2):trat+x2 , reprexdata)
summary(reprexmodel)
#> 
#> Call:
#> glm.nb(formula = y ~ trat * C + x1 + I(x1^2) + x1:trat + I(x1^2):trat + 
#>     x2, data = reprexdata, init.theta = 1.125627749, link = log)
#> 
#> Deviance Residuals: 
#>     Min       1Q   Median       3Q      Max  
#> -3.2650  -0.9743  -0.0763   0.3622   1.6155  
#> 
#> Coefficients:
#>                    Estimate Std. Error z value Pr(>|z|)    
#> (Intercept)         5.38260    0.39732  13.547   <2e-16 ***
#> tratFERTIL          0.82711    0.53994   1.532   0.1256    
#> CX1043             -0.24964    0.45297  -0.551   0.5816    
#> CX2062              0.69756    0.46684   1.494   0.1351    
#> x1                 -0.30131    0.27501  -1.096   0.2732    
#> I(x1^2)             0.07622    0.26750   0.285   0.7757    
#> x2                  0.49624    0.20320   2.442   0.0146 *  
#> tratFERTIL:CX1043  -0.92028    0.59184  -1.555   0.1200    
#> tratFERTIL:CX2062  -0.29311    0.61297  -0.478   0.6325    
#> tratFERTIL:x1      -0.08964    0.28592  -0.314   0.7539    
#> tratFERTIL:I(x1^2) -0.74654    0.33274  -2.244   0.0249 *  
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> 
#> (Dispersion parameter for Negative Binomial(1.1256) family taken to be 1)
#> 
#>     Null deviance: 107.153  on 63  degrees of freedom
#> Residual deviance:  74.949  on 53  degrees of freedom
#> AIC: 834.92
#> 
#> Number of Fisher Scoring iterations: 1
#> 
#> 
#>               Theta:  1.126 
#>           Std. Err.:  0.190 
#> 
#>  2 x log-likelihood:  -810.920
plot(ggeffect(reprexmodel, c("x1")))
```

![](https://i.imgur.com/sXIea4c.png)

``` r
plot(ggeffect(reprexmodel, c("x2")))
```

![](https://i.imgur.com/LGH17Of.png)

``` r
plot(ggeffect(reprexmodel, c("x1", "trat")))
```

![](https://i.imgur.com/LCGY3X1.png)

<sup>Created on 2018-11-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Best regards, 
Asier

**Comment by strengejacke** [comment]
Thanks, fixed! Will commit later...

**Comment by asierrl** [comment]
Thanks a lot. However, I'm not able to use the changed packages. 
Following the installing instructions I run the following:
library(githubinstall)
githubinstall::githubinstall("ggeffects")
library(ggeffects)

I got the following error message after the library(ggeffects) command:
Error: package or namespace load failed for ‘ggeffects’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 namespace ‘rlang’ 0.2.0 is already loaded, but >= 0.3.0 is required

However, it seems I have installed rlang 0.3.0.1:

> packageVersion("rlang")
[1] ‘0.3.0.1’

Am I missing something?
Best regards, 
Asier
